compiler/src/printer.ml

```diff
@@ -243,17 +244,27 @@ let pp_var_decl pp_var pp_size fmt v =
 
 let pp_gfun pp_info (pp_size:F.formatter -> 'size -> unit) pp_var fmt fd =
   let pp_vd =  pp_var_decl pp_var pp_size in
-(*  let locals = locals fd in *)
+  let pp_locals fmt fd =
+    let seen = ref Spv.empty in
+    let mark x = seen := Spv.add x !seen in
+    let is_seen x = Spv.mem x !seen in
+    List.iter mark fd.f_args;
+    fold_vars_fc (fun x () ->
+        if x.v_kind <> Const && not (is_seen x) then (
+          mark x;
+          F.fprintf fmt "%a;@ " pp_vd x)
+    ) () fd
+  in
```

Comment:
This is the only really non-trivial part of this PR. Could you add a comment explaining why this is needed please?

Reply:
Currently, the printer does not print the declaration of local variables (which is not correct). This snippet iters on all variables that are used in the function and prints its declaration (at most once, hence the `seen` set) unless it is a function argument (line 251) or a parameter (`v_kind <> Const`, line 253).

Reply:
Thanks. Does this mean that the ideal fix is to directly print the local declarations, or is your fix already the right solution?

Reply:
What do you call “directly print the local declarations”? There are no declarations of local variables in Jasmin abstract syntax.

Reply:
Oh, I didn't know that. Actually, I had the impression that in some places we have them and in some others we don't, but probably I'm wrong. Do you know why this choice was made? Does it make things simpler?

Reply:
When the set of local variables is needed (like here) we can compute it. Maybe this computation might be memoized?

At least we never worry about the consistency of this set (are all variables declared? are there variables that are declared but not used?).

Reply:
Thanks for the explanation!